scene/resources/material.cpp

```diff
@@ -1062,12 +1062,19 @@ void BaseMaterial3D::_update_shader() {
 		code += "	TANGENT = vec3(0.0,0.0,-1.0) * abs(NORMAL.x);\n";
 		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.y);\n";
 		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.z);\n";
-		code += "	TANGENT = normalize(TANGENT);\n";
-
+		if (flags[FLAG_UV1_USE_WORLD_TRIPLANAR]) {
+			code += "	TANGENT = normalize(TANGENT) * MODEL_NORMAL_MATRIX;\n";
+		} else {
+			code += "	TANGENT = normalize(TANGENT);\n";
+		}
```

Comment:
```suggestion
		}

```
Leave an empty line after this to group

Reply:
Why?

Reply:
Because it's a separate group of values, and you removed such an existing line in the original code, it was:
```cpp
		code += "	TANGENT = vec3(0.0,0.0,-1.0) * abs(NORMAL.x);\n";
		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.y);\n";
		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.z);\n";
		code += "	TANGENT = normalize(TANGENT);\n";

		code += "	BINORMAL = vec3(0.0,1.0,0.0) * abs(NORMAL.x);\n";
		code += "	BINORMAL+= vec3(0.0,0.0,-1.0) * abs(NORMAL.y);\n";
		code += "	BINORMAL+= vec3(0.0,1.0,0.0) * abs(NORMAL.z);\n";
		code += "	BINORMAL = normalize(BINORMAL);\n";
```

Much easier to read with the space

Reply:
Thanks for the explanation, I'll fix it!